instrumentation/opentelemetry-instrumentation-asgi/src/opentelemetry/instrumentation/asgi/__init__.py

```diff
@@ -342,6 +384,13 @@ async def __call__(self, scope, receive, send):
                     for key, value in attributes.items():
                         current_span.set_attribute(key, value)
 
+                    if span.kind == trace.SpanKind.SERVER:
```

Comment:
Please check for `span.is_recording()` before accessing `span.kind`. This causes exceptions to be raised if a `NonRecordingSpan` is received here.

It would be great to add a test to this instrumentation to ensure it doesn't crash when the SDK isn't loaded, see https://github.com/open-telemetry/opentelemetry-python-contrib/issues/958 for more details.

Reply:
I have changed `span` to `current_span` which is already being recording. So there will not be any exceptions.